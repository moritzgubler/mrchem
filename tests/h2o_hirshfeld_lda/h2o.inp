world_prec = 1.0e-4
world_unit = angstrom

WaveFunction {
  method = lda
}

SCF {
run = false
guess_type = GTO
guess_screen = 16.0
}

Molecule {
translate = true
$coords
O       0.0000  0.0000  -0.1250
H      -1.4375  0.0000   1.0250
H       1.4375  0.0000   1.0250
$end
}

Properties {
  hirshfeld_charges = true
}
