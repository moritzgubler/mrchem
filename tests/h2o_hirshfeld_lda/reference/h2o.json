{
  "input": {
    "constants": {
      "N_a": 6.02214076e+23,
      "angstrom2bohrs": 1.8897261246257702,
      "boltzmann_constant": 1.380649e-23,
      "dipmom_au2debye": 2.5417464739297717,
      "e0": 8.8541878128e-12,
      "electron_g_factor": -2.00231930436256,
      "elementary_charge": 1.602176634e-19,
      "fine_structure_constant": 0.0072973525693,
      "hartree2ev": 27.211386245988,
      "hartree2kcalmol": 627.5094740630558,
      "hartree2kjmol": 2625.4996394798254,
      "hartree2simagnetizability": 78.9451185,
      "hartree2wavenumbers": 219474.6313632,
      "light_speed": 137.035999084,
      "meter2bohr": 18897261246.2577
    },
    "geom_opt": {
      "init_step_size": -0.5,
      "max_force_component": 0.005,
      "max_history_length": 10,
      "max_iter": 100,
      "minimal_step_size": 0.01,
      "run": false,
      "subspace_tolerance": 0.001,
      "use_previous_guess": false
    },
    "molecule": {
      "charge": 0,
      "coords": [
        {
          "atom": "o",
          "r_rms": 5.0580178957e-05,
          "xyz": [
            0.0,
            0.0,
            -0.24317580771029856
          ]
        },
        {
          "atom": "h",
          "r_rms": 2.6569547399e-05,
          "xyz": [
            -2.7164813041495446,
            0.0,
            1.9300092356093368
          ]
        },
        {
          "atom": "h",
          "r_rms": 2.6569547399e-05,
          "xyz": [
            2.7164813041495446,
            0.0,
            1.9300092356093368
          ]
        }
      ],
      "multiplicity": 1
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": false,
      "omp_threads": -1,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 6,
      "basis_type": "interpolating",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -5
    },
    "printer": {
      "file_name": "h2o",
      "print_constants": false,
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.0001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.0001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "nuclear_model": "point_like",
          "proj_prec": 0.0001,
          "shared_memory": false,
          "smooth_prec": 0.0001
        },
        "xc_operator": {
          "shared_memory": false,
          "xc_functional": {
            "cutoff": 0.0,
            "functionals": [
              {
                "coef": 1.0,
                "name": "svwn5"
              }
            ],
            "spin": false
          }
        }
      },
      "initial_guess": {
        "environment": "None",
        "external_field": "None",
        "file_CUBE_a": "cube_vectors/CUBE_a_vector.json",
        "file_CUBE_b": "cube_vectors/CUBE_b_vector.json",
        "file_CUBE_p": "cube_vectors/CUBE_p_vector.json",
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "DFT (SVWN5)",
        "prec": 0.001,
        "relativity": "None",
        "restricted": true,
        "rotate": true,
        "screen": 16.0,
        "type": "gto",
        "zeta": 0
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.0001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        },
        "hirshfeld_charges": {
          "hirshfeld-1": {
            "precision": 0.0001
          }
        }
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        -5.341976807564934e-18,
        0.0,
        -4.4424263363005454e-17
      ],
      "charge": 0,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 1.6438995746893366,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            7.54141469758662e-07,
            -8.797188200272936e-12,
            1.6438995746891636
          ],
          "vector_el": [
            7.541484348538293e-07,
            0.0,
            -0.2707124348527402
          ],
          "vector_nuc": [
            -6.965095167288382e-12,
            -8.789230986014012e-12,
            1.914612009541904
          ]
        }
      },
      "geometry": [
        {
          "symbol": "O",
          "xyz": [
            0.0,
            0.0,
            -0.24317580771029856
          ]
        },
        {
          "symbol": "H",
          "xyz": [
            -2.7164813041495446,
            0.0,
            1.9300092356093368
          ]
        },
        {
          "symbol": "H",
          "xyz": [
            2.7164813041495446,
            0.0,
            1.9300092356093368
          ]
        }
      ],
      "hirshfeld_charges": {
        "hirshfeld-1": {
          "total": [
            -1.2256509183213495,
            0.6128254383501298,
            0.6128255933129395
          ]
        }
      },
      "multiplicity": 1,
      "orbital_energies": {
        "energy": [
          -5.831541422026438,
          0.2868881749197759,
          0.35041507983376347,
          0.4278104311862625,
          2.458354368701725
        ],
        "occupation": [
          2.0,
          2.0,
          2.0,
          2.0,
          2.0
        ],
        "spin": [
          "p",
          "p",
          "p",
          "p",
          "p"
        ],
        "sum_occupied": -4.616146734769822
      },
      "scf_energy": {
        "E_ee": 46.76254664433691,
        "E_eext": 0.0,
        "E_el": -48.595421761243195,
        "E_en": -162.42893318609498,
        "E_kin": 75.84734150946595,
        "E_next": 0.0,
        "E_nn": 4.783357401076065,
        "E_nuc": 4.783357401076065,
        "E_tot": -43.81206436016713,
        "E_x": 0.0,
        "E_xc": -8.77637672895107,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 8,
      "routine": "mrchem.x",
      "total_cores": 8,
      "version": "1.2.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 46.76254664433691,
        "E_eext": 0.0,
        "E_el": -48.595421761243195,
        "E_en": -162.42893318609498,
        "E_kin": 75.84734150946595,
        "E_next": 0.0,
        "E_nn": 4.783357401076065,
        "E_nuc": 4.783357401076065,
        "E_tot": -43.81206436016713,
        "E_x": 0.0,
        "E_xc": -8.77637672895107,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
