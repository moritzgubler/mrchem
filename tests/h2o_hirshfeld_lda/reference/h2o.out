                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                                                                     ***
***          __  __ ____   ____ _                                       ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                     ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                    ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |                   ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                   ***
***                                                                     ***
***         VERSION            1.2.0-alpha                              ***
***                                                                     ***
***         Git branch         property/hirshfeld                       ***
***         Git commit hash    13ea821841ed585ecac1                     ***
***         Git commit author  moritzgubler                             ***
***         Git commit date    Wed Sep 25 16:32:04 2024 +0200           ***
***                                                                     ***
***         Contact: luca.frediani@uit.no                               ***
***                                                                     ***
***         Radovan Bast            Magnar Bjorgve                      ***
***         Roberto Di Remigio      Antoine Durdek                      ***
***         Luca Frediani           Gabriel Gerez                       ***
***         Stig Rune Jensen        Jonas Juselius                      ***
***         Rune Monstad            Peter Wind                          ***
***                                                                     ***
***************************************************************************

---------------------------------------------------------------------------

 MPI processes           :       (no bank)                               1
 OpenMP threads          :                                               8
 Total cores             :                                               8
                                                                           
---------------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

---------------------------------------------------------------------------

 MRCPP version         : 1.6.0-alpha
 Git branch            : HEAD
 Git commit hash       : 720133372c9717134c5a
 Git commit author     : gitpeterwind
 Git commit date       : Thu Feb 15 13:44:31 2024 +0100

 Linear algebra        : EIGEN v3.4.0
 Parallelization       : MPI/OpenMP (8 threads)

---------------------------------------------------------------------------


                                                                           
===========================================================================
                         MultiResolution Analysis
---------------------------------------------------------------------------
 polynomial order      : 6
 polynomial type       : Interpolating
---------------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   32.00000    32.00000    32.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -32.00000   -32.00000   -32.00000 ]
 upper bounds          : [   32.00000    32.00000    32.00000 ]
 total length          : [   64.00000    64.00000    64.00000 ]
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                        Initializing Molecule                        ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0       O            :        0.000000        0.000000       -0.243176
    1       H            :       -2.716481        0.000000        1.930009
    2       H            :        2.716481        0.000000        1.930009
---------------------------------------------------------------------------
 Center of mass          :       -0.000000        0.000000       -0.000000
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                Computing Initial Guess Wavefunction                 ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial orbitals
 Method                  : Project GTO molecular orbitals
 Precision               : 1.00000e-03
 Screening               : 1.60000e+01 StdDev
 Restricted              : True
 MO file                 : initial_guess/mrchem.mop
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                            Molecular Orbitals
---------------------------------------------------------------------------
 Alpha electrons         :                                               5
 Beta electrons          :                                               5
 Total electrons         :                                              10
---------------------------------------------------------------------------
    n  Occ Spin          :                                            Norm
---------------------------------------------------------------------------
    0    2    p          :                              1.000000203320e+00
    1    2    p          :                              1.000000064900e+00
    2    2    p          :                              9.999999960131e-01
    3    2    p          :                              1.000000004599e+00
    4    2    p          :                              9.999999947944e-01
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial energy
 Method                  : DFT (SVWN5)
 Relativity              : None
 Environment             : None
 External fields         : None
 Precision               : 1.00000e-03
 Localization            : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                        Molecular Energy (initial)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                 75.847341509466
 E-N energy              :            (au)               -162.428933186095
 Coulomb energy          :            (au)                 46.762546644337
 Exchange energy         :            (au)                  0.000000000000
 X-C energy              :            (au)                 -8.776376728951
 N-N energy              :            (au)                  4.783357401076
---------------------------------------------------------------------------
 Electronic energy       :            (au)                -48.595421761243
 Nuclear energy          :            (au)                  4.783357401076
---------------------------------------------------------------------------
 Total energy            :            (au)             -4.381206436017e+01
                         :      (kcal/mol)             -2.749248546427e+04
                         :        (kJ/mol)             -1.150285591825e+05
                         :            (eV)             -1.192187005539e+03
===========================================================================


===========================================================================
                        Orbital Energies (initial)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -5.831541422026
    1    2    p          :            (au)                  0.286888174920
    2    2    p          :            (au)                  0.350415079834
    3    2    p          :            (au)                  0.427810431186
    4    2    p          :            (au)                  2.458354368702
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.616146734770
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                    Printing Molecular Properties                    ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               0
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0       O            :        0.000000        0.000000       -0.243176
    1       H            :       -2.716481        0.000000        1.930009
    2       H            :        2.716481        0.000000        1.930009
---------------------------------------------------------------------------
 Center of mass          :       -0.000000        0.000000       -0.000000
===========================================================================


===========================================================================
                         Molecular Energy (final)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                 75.847341509466
 E-N energy              :            (au)               -162.428933186095
 Coulomb energy          :            (au)                 46.762546644337
 Exchange energy         :            (au)                  0.000000000000
 X-C energy              :            (au)                 -8.776376728951
 N-N energy              :            (au)                  4.783357401076
---------------------------------------------------------------------------
 Electronic energy       :            (au)                -48.595421761243
 Nuclear energy          :            (au)                  4.783357401076
---------------------------------------------------------------------------
 Total energy            :            (au)             -4.381206436017e+01
                         :      (kcal/mol)             -2.749248546427e+04
                         :        (kJ/mol)             -1.150285591825e+05
                         :            (eV)             -1.192187005539e+03
===========================================================================


===========================================================================
                         Orbital Energies (final)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -5.831541422026
    1    2    p          :            (au)                  0.286888174920
    2    2    p          :            (au)                  0.350415079834
    3    2    p          :            (au)                  0.427810431186
    4    2    p          :            (au)                  2.458354368702
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.616146734770
===========================================================================


===========================================================================
                           Dipole Moment (dip-1)
---------------------------------------------------------------------------
 r_O                     :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Electronic vector       :        0.000001       -0.000000       -0.270712
 Magnitude               :            (au)                        0.270712
                         :         (Debye)                        0.688082
---------------------------------------------------------------------------
 Nuclear vector          :       -0.000000       -0.000000        1.914612
 Magnitude               :            (au)                        1.914612
                         :         (Debye)                        4.866458
---------------------------------------------------------------------------
 Total vector            :        0.000001       -0.000000        1.643900
 Magnitude               :            (au)                        1.643900
                         :         (Debye)                        4.178376
===========================================================================


===========================================================================
                      Hirshfeld Charges (hirshfeld-1)
---------------------------------------------------------------------------
---------------------------------------------------------------------------
 Charge of atom 0        :                                       -1.225651
 Charge of atom 1        :                                        0.612825
 Charge of atom 2        :                                        0.612826
---------------------------------------------------------------------------
 Sum of Hirshfeld charges:            (au)                    1.133417e-07
===========================================================================
                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                            Exiting MRChem                           ***
***                                                                     ***
***                       Wall time :  0h  0m  2s                       ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
